File: tests/verify.py
import os
import re
import sys

# Define base directory
BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))

def check_file_exists(relative_path):
    full_path = os.path.join(BASE_DIR, relative_path.replace('/', os.sep))
    if os.path.exists(full_path):
        print(f"   [OK] Found: {relative_path}")
        return True
    else:
        print(f"   [FAIL] MISSING: {relative_path} (Expected at: {full_path})")
        return False

def parse_js_data(file_path):
    """
    Rudimentary parser to extract image and link paths from the JS object.
    We don't need full JS parsing, just regex to find keys.
    """
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()

    # Extract objects between { }
    # This is a simple heuristic and might fail on complex nested structures, but fine for this config
    matches = re.findall(r'\{(.*?)\}', content, re.DOTALL)
    
    entries = []
    for match in matches:
        if "image:" in match and "link:" in match:
            # Extract image
            img_match = re.search(r'image:\s*["\'](.*?)["\']', match)
            link_match = re.search(r'link:\s*["\'](.*?)["\']', match)
            
            if img_match and link_match:
                entries.append({
                    'image': img_match.group(1),
                    'link': link_match.group(1)
                })
    return entries

print("Starting Verification...")
errors = 0

# 1. Verify Data Config Integrity
data_js_path = os.path.join(BASE_DIR, 'js', 'data.js')
if not os.path.exists(data_js_path):
    print(f"CRITICAL: js/data.js not found at {data_js_path}")
    sys.exit(1)

team_members = parse_js_data(data_js_path)
print(f"Found {len(team_members)} team configurations.")

for i, member in enumerate(team_members):
    print(f"\nChecking Member #{i+1}")
    if not check_file_exists(member['image']):
        errors += 1
    if not check_file_exists(member['link']):
        errors += 1

# 2. Verify Core Assets
core_assets = [
    'css/main.css',
    'css/landing.css',
    'css/responsive.css',
    'js/app.js',
    'index.html',
    'team/template.html',
    'assets/documents/resume.pdf'
]

print("\nChecking Core Assets...")
for asset in core_assets:
    if not check_file_exists(asset):
        errors += 1

print("\n-------------------")
if errors == 0:
    print("SUCCESS: All checks passed.")
    sys.exit(0)
else:
    print(f"FAILURE: {errors} errors found.")
    sys.exit(1)

File: tests/verify.js
const fs = require('fs');
const path = require('path');

// Mock DOM environment to load data.js
global.document = {};
const dataFile = fs.readFileSync('./js/data.js', 'utf8');

// extracting the array manually since require wouldn't work easily with the const declaration in browser-side JS
// We'll use a simple eval context or regex to parse it for this test
let teamMembers = [];
try {
    // Quick hack to evaluate the file content to get the variable
    // This is safe here because we own the file
    eval(dataFile);
} catch (e) {
    console.error("❌ Failed to parse js/data.js:", e.message);
    process.exit(1);
}

console.log(`ℹ️  Found ${teamMembers.length} team members.`);

let errors = 0;

teamMembers.forEach(member => {
    console.log(`\nTesting Member: ${member.name} (${member.id})`);

    // 1. Check Image Access
    const imgPath = path.join(__dirname, member.image);
    if (fs.existsSync(imgPath)) {
        console.log(`   ✅ Image found: ${member.image}`);
    } else {
        console.error(`   ❌ Image MISSING: ${member.image}`);
        errors++;
    }

    // 2. Check Portfolio Link Access
    const linkPath = path.join(__dirname, member.link);
    if (fs.existsSync(linkPath)) {
        console.log(`   ✅ Portfolio Page found: ${member.link}`);
    } else {
        console.error(`   ❌ Portfolio Page MISSING: ${member.link}`);
        errors++;
    }
});

// 3. Check Global Assets
const globalAssets = [
    'css/main.css',
    'css/landing.css',
    'css/responsive.css',
    'js/app.js',
    'assets/documents/resume.pdf'
];

console.log('\nTesting Global Assets:');
globalAssets.forEach(asset => {
    if (fs.existsSync(path.join(__dirname, asset))) {
        console.log(`   ✅ Asset found: ${asset}`);
    } else {
        console.error(`   ❌ Asset MISSING: ${asset}`);
        errors++;
    }
});

console.log('\n-------------------');
if (errors === 0) {
    console.log('✅ TEST PASSED: All links and assets valid.');
    process.exit(0);
} else {
    console.error(`❌ TEST FAILED: ${errors} errors found.`);
    process.exit(1);
}
